File: Token.java
public class Token {
    public enum Kind {
        IDENTIFIER, EQ, LET, PRINT, COMMA, SEMICOLON, INPUT, FOR,
        TO, NEXT, IF, THEN, GOTO, GOSUB, RETURN, END, STRING_LITERAL,
        INTEGER_LITERAL, PLUS, MINUS, TIMES, DIV, LT, GT, L_PAREN,
        R_PAREN, COLON, EOL, ERROR
    }

    private final int columnNumber;
    private final int lineNumber;
    private final Kind kind;
    private final String text;

    public Token(Kind kind, String text, int lineNumber, int columnNumber) {
        this.kind = kind;
        this.text = text;
        this.lineNumber = lineNumber;
        this.columnNumber = columnNumber;
    }

    public int getColumnNumber() {
        return columnNumber;
    }

    public int getLineNumber() {
        return lineNumber;
    }

    public Kind getKind() {
        return kind;
    }

    public String getText() {
        return text;
    }

    public String toString() {
        return String.format("Token[%s, '%s', %d, %d]",
            kind, text, lineNumber, columnNumber);
    }

    public static Kind lookupKeyword(String spelling) {
        switch (spelling) {
            case "let": return Kind.LET;
            case "print": return Kind.PRINT;
            case "input": return Kind.INPUT;
            case "for": return Kind.FOR;
            case "to": return Kind.TO;
            case "next": return Kind.NEXT;
            case "if": return Kind.IF;
            case "then": return Kind.THEN;
            case "goto": return Kind.GOTO;
            case "gosub": return Kind.GOSUB;
            case "return": return Kind.RETURN;
            case "end": return Kind.END;
            default: return null;
        }
    }

    public static Kind lookupSymbol(char c) {
        switch (c) {
            case '=': return Kind.EQ;
            case ',': return Kind.COMMA;
            case ';': return Kind.SEMICOLON;
            case ':': return Kind.COLON;
            case '+': return Kind.PLUS;
            case '-': return Kind.MINUS;
            case '*': return Kind.TIMES;
            case '/': return Kind.DIV;
            case '<': return Kind.LT;
            case '>': return Kind.GT;
            case '(': return Kind.L_PAREN;
            case ')': return Kind.R_PAREN;
            default: return null;
        }
    }
}

File: Interpreter/Token.cs
namespace Basic
{
    public record Token(TokenType Type, string Text, int Line, int Column)
    {
        public static TokenType? LookupKeyword(string text)
        {
            switch (text)
            {
                case "let": return TokenType.Let;
                case "print": return TokenType.Print;
                case "input": return TokenType.Input;
                case "for": return TokenType.For;
                case "to": return TokenType.To;
                case "next": return TokenType.Next;
                case "if": return TokenType.If;
                case "then": return TokenType.Then;
                case "goto": return TokenType.Goto;
                case "gosub": return TokenType.Gosub;
                case "return": return TokenType.Return;
                case "end": return TokenType.End;
                default: return null;
            }
        }

        public static TokenType? LookupSymbol(char c)
        {
            switch (c)
            {
                case '=': return TokenType.Eq;
                case ',': return TokenType.Comma;
                case ';': return TokenType.Semicolon;
                case ':': return TokenType.Colon;
                case '+': return TokenType.Plus;
                case '-': return TokenType.Minus;
                case '*': return TokenType.Mult;
                case '/': return TokenType.Div;
                case '%': return TokenType.Mod;
                case '<': return TokenType.Lt;
                case '>': return TokenType.Gt;
                case '(': return TokenType.ParenL;
                case ')': return TokenType.ParenR;
                default: return null;
            }
        }
    }
    public enum TokenType
    {
        Identifier,
        Eq,
        Let,
        Print,
        Comma,
        Semicolon,
        Input,
        For,
        To,
        Next,
        If,
        Then,
        Goto,
        Gosub,
        Return,
        End,
        StringLiteral,
        IntegerLiteral,
        Plus,
        Minus,
        Mult,
        Div,
        Mod,
        Lt,
        Gt,
        ParenL,
        ParenR,
        Colon,
        Eol
    }
}